### PR TITLE
Proposed lint rule change and fix for readme line length

### PR DESCRIPTION
**Pull request by subfuzion**
Feel free to close PR to reject proposal.

### CODE REVIEW

**Comment by kevmoo** [comment]
Is fine. Many files to cleanup, though!

**Comment by subfuzion** [comment]
> Is fine. Many files to cleanup, though!

Depends on lint rules, I guess. What rules would you like to enable? This is the only one that shows up for me with the current config (`tool/markdown_linter_config.yaml`)...

**Comment by kevmoo** [comment]
I'm cool w/ linting line length – just a lot things to fix

**Comment by subfuzion** [comment]
> I'm cool w/ linting line length – just a lot things to fix

Huh, I see that ... wonder why CL linting is showing different results than linting on the command line with the config under `tool`:

```
markdownlint -c tool/markdown_linter_config.yaml **/*.md
```

**Comment by kevmoo** [comment]
> > I'm cool w/ linting line length – just a lot things to fix
> 
> Huh, I see that ... wonder why CL linting is showing different results than linting on the command line with the config under `tool`:
> 
> ```
> markdownlint -c tool/markdown_linter_config.yaml **/*.md
> ```

Uh...I wonder if this action I have is using an old version of the markdown linter!

I've seen a few differences, too!

**Comment by kevmoo** [comment]
Ha!

https://github.com/GoogleCloudPlatform/functions-framework-dart/runs/1583062755?check_suite_focus=true#step:2:30

`0.18.0` in this action

I have 0.26.0 on my machine!

Hrm...

**Comment by subfuzion** [comment]
I'll figure this out after I review your other PR. I see that the markdown
lint cli from the same author as the package catches these. But to use that
CLI, you need a dot config under the root; doesn't seem to be any way to
specify under the `tool` directory.

`tool/markdown_linter_config.yaml`  ->  `.markdownlint.yml`

We can forget about it for now, or else consider moving in this direction
to perhaps make it easier to catch lint issues *before* CI.

**Comment by subfuzion** [comment]
Oops, crossed signals here, but in any case, lemme know what you think.

**Comment by subfuzion** [comment]
@kevmoo Proposed solution. My Github Action-fu is weak, so not sure I got this right. Basically proposing switching linter to this CLI from the same author as the linting package itself ("markdownlint-cli2 is designed to work well in conjunction with vscode-markdownlint.").

Works well locally:

```shell
markdownlint-cli2 **/*.md
```

PR includes all the recommended fixes based on lint rules. Note that line lengths greater than 80 are allowed if there are no whitespaces (like with long links).

File: .github/workflows/markdown_linter.yml [hunk 1]
```diff
@@ -10,17 +10,31 @@ on:
 jobs:
   markdown-link-check:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [14.x]
     steps:
     - uses: actions/checkout@master
-    - uses: gaurav-nelson/github-action-markdown-link-check@v1
```
Comment (kevmoo) [inline .github/workflows/markdown_linter.yml:15]: why did you remove this? Copy-past mistake?

File: .markdownlint.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+line_length:              # MD013 max line length
```
Comment (kevmoo) [inline .markdownlint.yaml:1]: Since this is yaml, put a comment header at the top explaining what this is and how it's used?

**Review by kevmoo — CHANGES_REQUESTED** [review]

File: .github/workflows/markdown_linter.yml (hunk 1 above)
Comment (subfuzion) [reply .github/workflows/markdown_linter.yml:15]: No, trying to use a different linter. I might not be doing this correctly, however. Mentioned in the comment above, probably need your help on this.

File: .markdownlint.yaml (hunk 2 above)
Comment (subfuzion) [reply .markdownlint.yaml:1]: No problem. Worth mentioning that the existing yaml doesn't have one either: https://github.com/GoogleCloudPlatform/functions-framework-dart/blob/main/tool/markdown_linter_config.yaml

Comment (subfuzion) [reply .markdownlint.yaml:7]: Done.

File: .github/workflows/markdown_linter.yml (hunk 1 above)
Comment (subfuzion) [reply .github/workflows/markdown_linter.yml:15]: Reverted.

Comment (subfuzion) [reply .github/workflows/markdown_linter.yml:15]: I just realized that I was misreading the job name -- I thought it said `markdown-lint-check`, not `markdown-link-check`!

Comment (subfuzion) [reply .github/workflows/markdown_linter.yml:15]: @kevmoo PTAL again. Fixed.

**Comment by kevmoo** [comment]
Here's the CRAZY thing

I run this locally and get

```shell
markdownlint-cli2-fix **/*.md ; echo $?
markdownlint-cli2 v0.0.13 (markdownlint v0.22.0)
Finding: CONTRIBUTING.md README.md docs/01-introduction.md docs/README.md docs/quickstarts/01-quickstart-dart.md docs/quickstarts/02-quickstart-docker.md docs/quickstarts/03-quickstart-cloudrun.md docs/release_process.md examples/hello/README.md examples/json/README.md examples/raw_cloudevent/README.md functions_framework/CHANGELOG.md functions_framework/README.md functions_framework_builder/CHANGELOG.md functions_framework_builder/README.md
Linting: 15 file(s)
Summary: 36 error(s)
CONTRIBUTING.md:62:81 MD013/line-length Line length [Expected: 80; Actual: 158]
CONTRIBUTING.md:68:81 MD013/line-length Line length [Expected: 80; Actual: 223]
docs/quickstarts/01-quickstart-dart.md:112:81 MD013/line-length Line length [Expected: 80; Actual: 206]
docs/quickstarts/01-quickstart-dart.md:159:81 MD013/line-length Line length [Expected: 80; Actual: 87]
docs/quickstarts/02-quickstart-docker.md:55:81 MD013/line-length Line length [Expected: 80; Actual: 137]
docs/quickstarts/02-quickstart-docker.md:56:81 MD013/line-length Line length [Expected: 80; Actual: 137]
docs/quickstarts/02-quickstart-docker.md:57:81 MD013/line-length Line length [Expected: 80; Actual: 137]
docs/quickstarts/02-quickstart-docker.md:58:81 MD013/line-length Line length [Expected: 80; Actual: 137]
docs/quickstarts/02-quickstart-docker.md:59:81 MD013/line-length Line length [Expected: 80; Actual: 137]
docs/quickstarts/02-quickstart-docker.md:60:81 MD013/line-length Line length [Expected: 80; Actual: 137]
docs/quickstarts/02-quickstart-docker.md:61:81 MD013/line-length Line length [Expected: 80; Actual: 137]
docs/quickstarts/02-quickstart-docker.md:62:81 MD013/line-length Line length [Expected: 80; Actual: 137]
docs/quickstarts/02-quickstart-docker.md:63:81 MD013/line-length Line length [Expected: 80; Actual: 137]
docs/quickstarts/02-quickstart-docker.md:64:81 MD013/line-length Line length [Expected: 80; Actual: 137]
docs/quickstarts/02-quickstart-docker.md:65:81 MD013/line-length Line length [Expected: 80; Actual: 137]
docs/quickstarts/02-quickstart-docker.md:66:81 MD013/line-length Line length [Expected: 80; Actual: 137]
docs/quickstarts/02-quickstart-docker.md:67:81 MD013/line-length Line length [Expected: 80; Actual: 137]
docs/quickstarts/02-quickstart-docker.md:68:81 MD013/line-length Line length [Expected: 80; Actual: 137]
docs/quickstarts/02-quickstart-docker.md:69:81 MD013/line-length Line length [Expected: 80; Actual: 137]
docs/quickstarts/02-quickstart-docker.md:70:81 MD013/line-length Line length [Expected: 80; Actual: 137]
docs/quickstarts/02-quickstart-docker.md:71:81 MD013/line-length Line length [Expected: 80; Actual: 137]
docs/quickstarts/02-quickstart-docker.md:72:81 MD013/line-length Line length [Expected: 80; Actual: 137]
docs/quickstarts/02-quickstart-docker.md:73:81 MD013/line-length Line length [Expected: 80; Actual: 137]
docs/quickstarts/02-quickstart-docker.md:74:81 MD013/line-length Line length [Expected: 80; Actual: 137]
docs/quickstarts/02-quickstart-docker.md:75:81 MD013/line-length Line length [Expected: 80; Actual: 137]
docs/quickstarts/02-quickstart-docker.md:76:81 MD013/line-length Line length [Expected: 80; Actual: 137]
docs/quickstarts/02-quickstart-docker.md:77:81 MD013/line-length Line length [Expected: 80; Actual: 137]
docs/quickstarts/02-quickstart-docker.md:199:81 MD013/line-length Line length [Expected: 80; Actual: 89]
docs/quickstarts/03-quickstart-cloudrun.md:82:81 MD013/line-length Line length [Expected: 80; Actual: 122]
docs/quickstarts/03-quickstart-cloudrun.md:85:81 MD013/line-length Line length [Expected: 80; Actual: 160]
docs/quickstarts/03-quickstart-cloudrun.md:86:81 MD013/line-length Line length [Expected: 80; Actual: 89]
docs/quickstarts/03-quickstart-cloudrun.md:90:81 MD013/line-length Line length [Expected: 80; Actual: 99]
examples/json/README.md:128:81 MD013/line-length Line length [Expected: 80; Actual: 93]
examples/raw_cloudevent/README.md:87:81 MD013/line-length Line length [Expected: 80; Actual: 109]
README.md:15:81 MD013/line-length Line length [Expected: 80; Actual: 90]
README.md:53:81 MD013/line-length Line length [Expected: 80; Actual: 82]
1
```


🤷

**Comment by kevmoo** [comment]
VS

```shell
Run markdownlint-cli2-fix **/*.md
  markdownlint-cli2-fix **/*.md
  shell: /bin/bash -e {0}
markdownlint-cli2 v0.0.13 (markdownlint v0.22.0)
Finding: docs/01-introduction.md docs/README.md docs/release_process.md functions_framework/CHANGELOG.md functions_framework/README.md functions_framework_builder/CHANGELOG.md functions_framework_builder/README.md
Linting: 7 file(s)
Summary: 0 error(s)
```

At https://github.com/GoogleCloudPlatform/functions-framework-dart/runs/1585295259?check_suite_focus=true#step:4:1

🤦

**Comment by subfuzion** [comment]
> ```shell
> markdownlint-cli2-fix **/*.md ; echo $?
> ```

Very strange -- not what I see:

```
$ markdownlint-cli2-fix **/*.md ; echo $?
markdownlint-cli2 v0.0.13 (markdownlint v0.22.0)
Finding: docs/01-introduction.md docs/README.md docs/release_process.md functions_framework/CHANGELOG.md functions_framework/README.md functions_framework_builder/CHANGELOG.md functions_framework_builder/README.md
Linting: 7 file(s)
Summary: 0 error(s)
0
```

And we're on the same version.

**Comment by subfuzion** [comment]
Make sure you're testing on this PR branch, though -- because I took the time to fix all the lint errors. You're maybe in a different (unfixed) branch?

**Comment by subfuzion** [comment]
Oh I see -- you are in the `docs` branch. You need to run from the repo root (so that the CLI can find its config file (`.markdownlint.yaml`).

**Review by kevmoo — APPROVED** [review]
:shipit:

**Comment by kevmoo** [comment]
Just make sure to squash 😄

**Comment by subfuzion** [comment]
lol -- we disabled all the other options.
